File: api/ShiftGiving/Models/Campaign.cs
namespace ShiftGiving.Models;

public class Campaign
{
    public Guid Id { get; set; }
    public Guid OrganizationId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? ShortDescription { get; set; }
    public decimal GoalAmount { get; set; }
    public decimal RaisedAmount { get; set; }
    public CampaignStatus Status { get; set; } = CampaignStatus.Draft;
    public DateOnly StartDate { get; set; }
    public DateOnly EndDate { get; set; }
    public string? FeaturedImageUrl { get; set; }
    public string? VideoUrl { get; set; }
    public string? SocialFacebook { get; set; }
    public string? SocialTwitter { get; set; }
    public string? SocialInstagram { get; set; }
    public string? SocialLinkedin { get; set; }
    public bool IsFeatured { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Organization Organization { get; set; } = null!;
    public ICollection<Donation> Donations { get; set; } = new List<Donation>();
    public ICollection<CampaignImage> Images { get; set; } = new List<CampaignImage>();
    public ICollection<CampaignProgram> Programs { get; set; } = new List<CampaignProgram>();
}

public enum CampaignStatus
{
    Draft,
    Active,
    ClosingSoon,
    Completed,
    Cancelled
}

File: mobile/lib/data/models/campaign.dart
class Campaign {
  final String id;
  final String title;
  final String? shortDescription;
  final double goalAmount;
  final double raisedAmount;
  final String status;
  final String? featuredImageUrl;
  final OrganizationBasic organization;
  final DateTime endDate;

  Campaign({
    required this.id,
    required this.title,
    this.shortDescription,
    required this.goalAmount,
    required this.raisedAmount,
    required this.status,
    this.featuredImageUrl,
    required this.organization,
    required this.endDate,
  });

  factory Campaign.fromJson(Map<String, dynamic> json) {
    return Campaign(
      id: json['id'] as String,
      title: json['title'] as String,
      shortDescription: json['shortDescription'] as String?,
      goalAmount: (json['goalAmount'] as num).toDouble(),
      raisedAmount: (json['raisedAmount'] as num).toDouble(),
      status: json['status'] as String,
      featuredImageUrl: json['featuredImageUrl'] as String?,
      organization: OrganizationBasic.fromJson(
        json['organization'] as Map<String, dynamic>,
      ),
      endDate: DateTime.parse(json['endDate'] as String),
    );
  }
}

class OrganizationBasic {
  final String id;
  final String name;
  final String? logoUrl;

  OrganizationBasic({
    required this.id,
    required this.name,
    this.logoUrl,
  });

  factory OrganizationBasic.fromJson(Map<String, dynamic> json) {
    return OrganizationBasic(
      id: json['id'] as String,
      name: json['name'] as String,
      logoUrl: json['logoUrl'] as String?,
    );
  }
}
